src/mito2/src/sst/file_purger.rs

```diff
@@ -35,11 +38,98 @@ pub trait FilePurger: Send + Sync {
 
 pub type FilePurgerRef = Arc<dyn FilePurger>;
 
-// TODO(yingwen): Remove this once we implement the real purger.
-/// A purger that does nothing.
-#[derive(Debug)]
-struct NoopPurger {}
+pub struct LocalFilePurger {
+    pub scheduler: Arc<LocalScheduler>,
+
+    pub sst_layer: Arc<AccessLayer>,
+}
+
+impl LocalFilePurger {
+    pub fn new(scheduler: Arc<LocalScheduler>, sst_layer: Arc<AccessLayer>) -> Self {
+        Self {
+            scheduler,
+            sst_layer,
+        }
+    }
+}
+
+impl FilePurger for LocalFilePurger {
+    fn send_request(&self, request: PurgeRequest) {
+        let file_id = request.file_id;
+        let region_id = request.region_id;
+        let sst_layer = self.sst_layer.clone();
+
+        let _ = self
+            .scheduler
+            .schedule(Box::pin(async move {
+                if let Err(e) = sst_layer.delete_sst(file_id).await {
+                    error!(e; "Failed to delete SST file, file: {}, region: {}", 
+                        file_id.as_parquet(), region_id);
+                } else {
+                    debug!(
+                        "Successfully deleted SST file: {}, region: {}",
+                        file_id.as_parquet(),
+                        region_id
+                    );
+                }
+            }))
+            .map_err(|e| {
+                error!(e; "Failed to schedule the file purge request");
+                e
+            });
```

Comment:
style: This would be nit and clean
```suggestion
        if let Err(e) = self
            .scheduler
            .schedule(Box::pin(async move {
                if let Err(e) = sst_layer.delete_sst(file_id).await {
                    error!(e; "Failed to delete SST file, file: {}, region: {}", 
                        file_id.as_parquet(), region_id);
                } else {
                    debug!(
                        "Successfully deleted SST file: {}, region: {}",
                        file_id.as_parquet(),
                        region_id
                    );
                }
            })) {
            error!(e; "Failed to schedule the file purge request");
        }
```

Reply:
👍